### PR TITLE
swap the training order in semisupervised_translation

### PR DESCRIPTION
**Pull request by hxbai**
it is better to train denoising autoencoder first.
for unsupervised mt, it may have better backtranslation.
ref: https://github.com/facebookresearch/XLM/blob/master/train.py#L288-L298

### CODE REVIEW

**Comment by hxbai** [comment]
Is it needed? @pipibjc

**Comment by pipibjc** [comment]
This won't change how we update the model. Unlike XLM, which update the model after each `trainer.xx_step`, fairseq implementation only update the model after we accumulate the gradients from all three tasks after each batch, i.e. [here](https://github.com/pytorch/fairseq/blob/master/fairseq/trainer.py#L341) which is after the `[train_step](https://github.com/pytorch/fairseq/blob/master/fairseq/trainer.py#L258-L261)`.

**Comment by pipibjc** [comment]
We also compared between 1) update the model after each task (XLM implementation), and 2) update the model after all three tasks (fairseq implementation), but we didn't find any significant difference. We can get comparable unsupervised fr-en unsupervised MT result in terms of BLEU.
The latter one is easier to implement under current fairseq framework, so we went with 2).

**Comment by hxbai** [comment]
Thank you! sorry for my carelessness. My experiments also show comparable results.

**Comment by pipibjc** [comment]
No worries. Thanks for bringing this up!

**Comment by Hbenmazi** [comment]
Hi! Could you tell me how to use fairseq for unsupervised translation?
